bt/behavior_tree.cpp

```
@@ -82,9 +83,15 @@ Ref<BTInstance> BehaviorTree::instantiate(Node *p_agent, const Ref<Blackboard> &
 	ERR_FAIL_COND_V_MSG(p_blackboard.is_null(), nullptr, "BehaviorTree: Instantiation failed - blackboard can't be null.");
 	Node *scene_root = p_custom_scene_root ? p_custom_scene_root : p_instance_owner->get_owner();
 	ERR_FAIL_NULL_V_MSG(scene_root, nullptr, "BehaviorTree: Instantiation failed - unable to establish scene root. This is likely due to the instance owner not being owned by a scene node and custom_scene_root being null.");
-	Ref<BTTask> root_copy = root_task->clone();
-	root_copy->initialize(p_agent, p_blackboard, scene_root);
-	return BTInstance::create(root_copy, get_path(), p_instance_owner);
+	Ref<BTTask> new_root;
+	if (root_task->is_enabled_in_tree()) {
```

I think `clone()` might return `nullptr` for various reasons, especially that it's a virtual method. Let’s check if the return value `is_null()` and just use the disabled task fallback if it is.

---

That could potentially hide bugs, no? Being disabled is a known case to handle, but if `clone()` returned null for another reason I would think it's preferable to get an error than to have a silently failing BT. Will defer to you.

---

Thoughts on this flow?
```
if clone == null {
  if root_task.enabled {
    push_error("oh no")
  }
  add_task(BTFail)
}
```